### DISCUSSION TITLE
表单的 disabled 和组件的 disabled 逻辑问题

### DISCUSSION THREAD

**Discussion by minosss** [ant-design/ant-design#44255] [General]
按道理整个表单是禁止的状态下，下面的组件也应该是禁止的吧？

目前我看到是自定义的 disabled 属性判断是这样的

```js
// form disabled
const disabled = React.useContext(DisabledContext);
const mergedDisabled = customDisabled ?? disabled;
```

也就是组件设置 `disabled=false` 的情况下 `<Form disabled />` 也一样无效

原本就是是这样设计的？

**Comment by Wxh16144** [comment]
设计就是组件自己本身的 disabled 优先级最高， 其次是组件如果在 Form 表单下，则会用 Form 组件的 disabled context。

表单是禁用状态，输入框则是非禁用状态
```tsx
<Form disabled>
  <Form.Item label="Input">
    <Input disabled={false} />
  </Form.Item>
</Form>
```

**Reply by minosss** [reply]
了解

**Reply by coldsmirk** [reply]
大佬们，我想问一下，自定义表单组件中我如何获取整个表单的禁用状态呢？DisabledContext 这个 antd 没有导出啊
